File: JavaScript/496. Next Greater Element I.js
/**
 * @param {number[]} nums1
 * @param {number[]} nums2
 * @return {number[]}
 */
var nextGreaterElement = function(nums1, nums2) {
    var a = [], b = 0;
    for(var i = 0;i < nums1.length;i++){
        b = nums1[i];
        for(var j = nums2.indexOf(b);j < nums2.length;j++){
            if(nums2[j] > b){
                a.push(nums2[j]);
                break;
            }
        }
        if(j == nums2.length) a.push(-1);
    }
    return a;
};

File: Python/496. Next Greater Element I.py
class Solution:
    def nextGreaterElement(self, nums1: List[int], nums2: List[int]) -> List[int]:
        result = []
        size2 = len(nums2)
        for i in nums1:
            for j in range(nums2.index(i), size2):
                if nums2[j] > i:
                    result.append(nums2[j])
                    break
                elif j == size2 - 1:
                    result.append(-1)
        return result
